File: 347-top-k-frequent-elements/top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        d= {}
        for i in nums:
            if i in d:
                d[i]+=1
            else:
                d[i]=1
        r=[]
        for i in range(k):
            k=max(d , key=d.get)
            r.append(k)
            del d[k]
        return r

File: 347-top-k-frequent-elements/top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> freq;
        for( int i :nums)
            freq[i]++;
        vector<pair<int,int>> vec(freq.begin(),freq.end());
        sort(vec.begin(),vec.end() ,[](auto &a , auto &b ){
            return a.second>b.second;
        });
        vector<int> fr;
        for(int i=0;i<k;i++)
            fr.push_back(vec[i].first);
        return fr;

        

    }
};